### PR TITLE
refactor(flat-table): update styled system implementation

### PR DESCRIPTION
**Pull request by mmaciorowski-sage**
BREAKING CHANGE: FlatTable uses margin. FlatTableCell FlatTableHeader FlatTableRowHeader use padding

### Proposed behaviour
`@styled-system/space` margin only (remove padding - breaking change).
Filter styled system margin props.

### Current behaviour
Uses no filtered `propTypes.space`.

### Checklist

- [x] Commits follow our style guide
- [ ] Screenshots are included in the PR if useful
- [ ] All themes are supported if required
- [ ] Unit tests added or updated if required
- [ ] Cypress automation tests added or updated if required
- [ ] Storybook added or updated if required
- [x] Typescript `d.ts` file added or updated if required
- [ ] Carbon implementation and Design System documentation are congruent

### Testing instructions
https://codesandbox.io/s/carbon-quickstart-forked-uc7mp

### CODE REVIEW

**Comment by edleeks87** [comment]
I think the description of the `BREAKING CHANGE` is wrong as some of the components have been updated to use the `padding` not `margin` interface etc

File: src/components/flat-table/flat-table-cell/flat-table-cell.style.js [hunk 1]
```diff
@@ -16,7 +16,7 @@ const StyledFlatTableCell = styled.td`
 
     > div {
```
Comment (edleeks87) [inline src/components/flat-table/flat-table-cell/flat-table-cell.style.js:17]: I don't think these will override the defaults applied for padding left and right (see screenshot). The unit tests pass as the styling is applied here https://github.com/Sage/carbon/blob/c547f4efdd035bb2dd0013b94c21bb38df8237c7/src/components/flat-table/flat-table.style.js#L56  I think you will need to raise the specificity like it is currently in `master` like so https://github.com/Sage/carbon/blob/6dba4e9ae48618e7d6ea9fbd6e9ffe6323aea730/src/components/flat-table/flat-table-cell/flat-table-cell.style.js#L29

image

File: src/components/flat-table/flat-table-header/flat-table-header.style.js [hunk 2]
```diff
@@ -30,7 +30,7 @@ const StyledFlatTableHeader = styled.th`
 
     > div {
```
Comment (edleeks87) [inline src/components/flat-table/flat-table-header/flat-table-header.style.js:31]: see comment about raising specificity

**Review by edleeks87 — CHANGES_REQUESTED** [review]
Just the commit message and the specificity issue, everything else looks good

**Comment by mmaciorowski-sage** [comment]
> I think the description of the `BREAKING CHANGE` is wrong as some of the components have been updated to use the `padding` not `margin` interface etc

Good point. Changed.

File: src/components/flat-table/flat-table-cell/flat-table-cell.style.js (hunk 1 above)
Comment (mmaciorowski-sage) [reply src/components/flat-table/flat-table-cell/flat-table-cell.style.js]: Good point, changed. Thanks!

**Review by edleeks87 — APPROVED** [review]

**Review by dawidzarzycki — APPROVED** [review]

**Review by DlgSHi — APPROVED** [review]

**Pull request merged**